Invalid PKCS8 data Exception while parsing fs.gs.auth.service.account.private.key

We are seeing this exception while parsing `fs.gs.auth.service.account.private.key` value if it contains `\n` instead of line break 

```
E   20/09/10 04:26:56  tools.DistCp: Invalid arguments: 
E   java.io.IOException: Invalid PKCS8 data.
E       at com.google.cloud.hadoop.repackaged.gcs.com.google.cloud.hadoop.util.CredentialFactory.privateKeyFromPkcs8(CredentialFactory.java:419)
E       at com.google.cloud.hadoop.repackaged.gcs.com.google.cloud.hadoop.util.CredentialFactory.getCredentialsFromSAParameters(CredentialFactory.java:324)
E       at com.google.cloud.hadoop.repackaged.gcs.com.google.cloud.hadoop.util.CredentialFactory.getCredential(CredentialFactory.java:481)
E       at com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemBase.getCredential(GoogleHadoopFileSystemBase.java:1443)
E       at com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemBase.createGcsFs(GoogleHadoopFileSystemBase.java:1510)
E       at com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemBase.configure(GoogleHadoopFileSystemBase.java:1486)
E       at com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemBase.initialize(GoogleHadoopFileSystemBase.java:541)
E       at com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemBase.initialize(GoogleHadoopFileSystemBase.java:494)
E       at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:33`
```

Example private key file 
`-----BEGIN RIVATE KEY-----\nMIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCxOyswh6/7bctP\ny2dIp0I2O0bmZxEr4ZH6bJ+ZTX9kf4HEj7l1QHO2lVws0qsEZarcxnKJyDLn+3De\n5ZJKwgd+VLvc4qrgRrTnN7/pxk1eyNAOzq8Kb6YjbS7BLwC+C6m9WiWpbb5VmbrR\nGRsI4gh96MarPsuXV7zRDae1OaEJkJZxH3OgwU4mWnIBc1X5QPEVebhd60y0bQMh\nlVXwqjMVBNJF4X76/njzVIy7/gYVjzOiBnk7rZH92aJFFpSqVb+oeFNAD3/2yw9Y\nw1dpQ1cX/gL12BBnEJ4P8PTiIIYsR3EWC/D6gC9PGkSt+GxsGuR/QTCp3y3AILv8\nmismMiptAgMBAAECggEADS70kICXY6GQLRhZT3s5nqUcKgliqpv16uXnnVS5dDVT\n5tY2GnW35c4IdEEbfZE+e3u7i2o79yuqArqa/jqeqhQNgWnPAi3gSQ7JlM8udC2F\n5IQVWzZg6mO1xaMEcB6X+sDY6liCfwWXcvRmIWwG667EEHr0VAavwfE9F2r5AqgC\nB8mZOofpvSPCEkBkoQKRLZSD98cR3mR5HupsEhOzYjA5R6iurS+F/LbL8ewzBN0s\nlzDRXhtsQH5/bAM7ob8gJ6FLV6MpN+truncated.........=\n-----END PRIVATE KEY-----\n`

As per debugging, I have found a workaround which is changing  `\n` into actual new lines.

Is this an expected behavior?? Please help. Thanks

--- a/util/src/main/java/com/google/cloud/hadoop/util/CredentialFactory.java
+++ b/util/src/main/java/com/google/cloud/hadoop/util/CredentialFactory.java
@@ -383,8 +383,8 @@ public class CredentialFactory {
 
   // TODO: Copied (mostly) over from Google Credential since it has private scope
   private static PrivateKey privateKeyFromPkcs8(String privateKeyPem) throws IOException {
-    privateKeyPem = privateKeyPem.replaceAll("\\n", System.lineSeparator());
-    Reader reader = new StringReader(privateKeyPem);
+    // Unescape new-line symbols in privateKeyPem string value
+    Reader reader = new StringReader(privateKeyPem.replaceAll("\\n", System.lineSeparator()));
     Section section = PemReader.readFirstSectionAndClose(reader, "PRIVATE KEY");
     if (section == null) {
       throw new IOException("Invalid PKCS8 data.");

--- a/util/src/main/java/com/google/cloud/hadoop/util/CredentialFactory.java
+++ b/util/src/main/java/com/google/cloud/hadoop/util/CredentialFactory.java
@@ -383,6 +383,7 @@ public class CredentialFactory {
 
   // TODO: Copied (mostly) over from Google Credential since it has private scope
   private static PrivateKey privateKeyFromPkcs8(String privateKeyPem) throws IOException {
+    privateKeyPem = privateKeyPem.replaceAll("\\n", System.lineSeparator());
     Reader reader = new StringReader(privateKeyPem);
     Section section = PemReader.readFirstSectionAndClose(reader, "PRIVATE KEY");
     if (section == null) {

